main.go

```diff
@@ -14,74 +9,44 @@ func main() {
 	flags := getCLflags()
 
 	conf := NewConfig()
-	log.Debug(conf.S3)
-
-	sb, err := newS3Backend(conf.S3)
+	pg := conf.db
+	log.Debug(conf.s3)
 
+	sb, err := newS3Backend(conf.s3)
 	if err != nil {
 		log.Fatal(err)
 	}
 
-	retryBackoff := backoff.NewExponentialBackOff()
-
-	tr := transportConfigES(conf.Elastic)
-
-	c, err := elastic.NewClient(elasticsearch.Config{
-		Addresses: []string{
-			flags.instance,
-		},
-		Username:      conf.Elastic.user,
-		Password:      conf.Elastic.password,
-		RetryOnStatus: []int{502, 503, 504, 429},
-		RetryBackoff: func(i int) time.Duration {
-			if i == 1 {
-				retryBackoff.Reset()
-			}
-			return retryBackoff.NextBackOff()
-		},
-		MaxRetries: 5,
-		Transport:  tr,
-	})
-
+	elastic, err := newElasticClient(conf.elastic)
 	if err != nil {
 		log.Fatal(err)
 	}
 
 	switch flags.action {
-	case "backup":
-		log.Infof("Loading index %s into %s", flags.indexName, flags.instance)
-		backupData(*sb, *c, conf.keyPath, flags.indexName, flags.batchsize)
-	case "restore":
-		err = countDocuments(*c, flags.indexName)
+	case "es_backup":
+		err := elastic.backupDocuments(sb, conf.keyPath, flags.name)
+		if err != nil {
+			log.Fatal(err)
+		}
+	case "es_restore":
+		err := elastic.restoreDocuments(sb, conf.keyPath, flags.name)
+		if err != nil {
+			log.Fatal(err)
+		}
+	case "es_create":
+		err := elastic.indexDocuments(flags.name)
 		if err != nil {
 			log.Fatal(err)
 		}
-		log.Infof("Dumping index %s into %s", flags.indexName, flags.instance)
-		restoreData(*sb, *c, conf.keyPath, flags.indexName, flags.batchsize)
-	case "create":
-		indexName := flags.indexName + "-" + "test"
-		log.Infof("Creating index %s in %s", indexName, flags.instance)
-		indexDocuments(*c, indexName)
 	case "pg_dump":
```

Comment:
Here we don't have semantic consistency. Note "backup" and "dump". Do we want it?

Reply:
You backup elasticsearch indexes, but you dump a database.
I had the idea to merge all so we only have `backup` and  `restore` regardless of what we do backup of.

Reply:
Yes that's good. So let's do that

Reply:
Had a look at this, it will require an even larger change, something best left util after Christmas.